Allowing boolean array in delayed_subsample

ameliefroessl opened the pull request:
This PR adds the capability of subsampling from boolean array inputs in `delayed_subsample`. 

 - [x] Adds feat: allowing boolean arrays,
 - [x] Tests added,
 - [ ] Fully documented, including `api/*.md` for new API.

ameliefroessl committed:
feat: allowing boolean array in delayed_subsample

rhugonnet reviewed (approved):
Perfect! We'd just have to mention the behaviour for boolean arrays in the function description, and add a test that checks it returns exactly the same subsample for a boolean array matching a NaN array, then we can merge https://github.com/GlacioHack/geoutils/pull/537!

ameliefroessl committed:
docs: add boolean note to delayed_subsample

ameliefroessl committed:
tests: add tests for boolean input array for delayed_subsample

ameliefroessl commented:
Thanks for the feedback @rhugonnet! I've updated the tests and updated the docstrings to reflect the changes. Let me know if anything is missing otherwise feel free to merge  :+1: :smiley:
